bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -177,6 +175,26 @@ uint64_t DbCheckpointManager::directorySize(const _fs::path& directory, const bo
   }
   return size;
 }
+void DbCheckpointManager::saveDbMetadataToReservedPages(SeqNum sn, std::chrono::seconds timestamp) {
+  shared_metadata_.lastCmdSeqNum_ = sn;
+  shared_metadata_.lastCmdTimestamp_ = timestamp;
+  std::ostringstream outStream;
+  concord::serialize::Serializable::serialize(outStream, shared_metadata_);
+  auto data = outStream.str();
+  saveReservedPage(0, data.size(), data.data());
+  lastCheckpointCreationTime_ = shared_metadata_.lastCmdTimestamp_;
+}
+
+void DbCheckpointManager::loadDbMetadataFromReservedPages() {
+  if (!loadReservedPage(0, sizeOfReservedPage(), page.data())) {
+    LOG_WARN(getLogger(), "unable to read db checkpoint shared metadata from reserved pages");
+    return;
+  }
+  std::istringstream inStream;
+  inStream.str(page);
+  concord::serialize::Serializable::deserialize(inStream, shared_metadata_);
+  lastCheckpointCreationTime_ = shared_metadata_.lastCmdTimestamp_;
```

Comment:
you don't load the sn from the reserved pages

Reply:
fixed